planner/core/planbuilder.go

```
@@ -1293,27 +1293,56 @@ func (b *planBuilder) buildSelectPlanOfInsert(insert *ast.InsertStmt, insertPlan
 	return nil
 }
 
+func (b *planBuilder) buildSetValuesOfLoadData(ld *ast.LoadDataStmt, ldPlan *LoadData, mockTablePlan *LogicalTableDual) error {
+	for _, assign := range ld.SetList {
+		exprCol, err := ldPlan.tableSchema.FindColumn(assign.Column)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if exprCol == nil {
+			return errors.Errorf("Can't find column %s", assign.Column)
```

It's better to return the same error as MySQL does.